lib/appsignal/probes/erlang_probe.ex

```diff
@@ -73,6 +74,21 @@ defmodule Appsignal.Probes.ErlangProbe do
     set_gauge("total_run_queue_lengths", total - cpu, %{type: "io"})
   end
 
+  defp scheduler_utilization_metrics(sample) do
+    unless is_nil(sample) do
+      utilization = :scheduler.utilization(sample)
+
+      utilization
+      |> Enum.map(&Tuple.to_list/1)
+      |> Enum.filter(fn [type | _] -> type == :normal end)
+      |> Enum.each(fn [_, id, value, _] ->
+        set_gauge("erlang_scheduler_utilization", value * 100, %{type: "scheduler", id: "#{id}"})
```

Comment:
The type tag is always hardcoded right now. Do we expect other types to be added in the future that we want to tag it in advance to keep them separate?

Reply:
That's a good question, I was wondering that as well. There _are_ other schedulers that we could send metrics about: from this very same `:scheduler.utilization/1` call, we also get utilization values for `type == :cpu` and `type == :io` schedulers, which we're currently ignoring. Possible action points that come to mind:

- Follow Erlang's naming here and set `type` to "normal" instead of "scheduler" (will need changing in appsignal/public_config#33 as well)
- Send `io` and `cpu` as well, even if we're not going to display them in the magic dashboard for now (there's a bunch of other values we send that aren't displayed in the magic dashboard either)
- Remove `type` entirely; if we change our mind later about the other schedulers, they could always be sent as separate metrics, named `erlang_io_scheduler_utilization` or something like that.

Reply:
> (will need changing in appsignal/public_config#33 as well)

Now that I look at it, we might as well change the hardcoded tag value "scheduler" to wildcard as well.

> There are other schedulers that we could send metrics about

If there are other ones we may add in the future the tag usage is fine like this. Let's create other issues for those metrics and discuss if we want to add them and when.

Reply:
I’d say this is fine for now, but we might want to change the type to “normal" to match Erlang’s conventions.

Reply:
Better change it now than later, then 👍